src/main/java/com/clickhouse/kafka/connect/transforms/KeyToValue.java

```diff
@@ -0,0 +1,81 @@
+package com.clickhouse.kafka.connect.transforms;
+
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.connect.connector.ConnectRecord;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaBuilder;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.transforms.Transformation;
+import org.apache.kafka.common.config.AbstractConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+
+public class KeyToValue<R extends ConnectRecord<R>> implements Transformation<R> {
+    private static final Logger LOGGER = LoggerFactory.getLogger(KeyToValue.class.getName());
+    public static final ConfigDef CONFIG_DEF = new ConfigDef().define("field", ConfigDef.Type.STRING, "_key", ConfigDef.Importance.LOW,
+                    "Field name on the record value to extract the record key into.");
+
+    private String keyFieldName;
+    private Schema valueSchema;
+
+    @Override
+    public void configure(Map<String, ?> configs) {
+        final SimpleConfig config = new SimpleConfig(CONFIG_DEF, configs);
+        keyFieldName = config.getString("field");
+    }
+
+    @Override
+    public R apply(R record) {
+        LOGGER.debug("Old Key: {}, Old Value: {}", record.key(), record.value());
+        if (record.valueSchema() == null) {
+            return applySchemaless(record);
+        } else {
+            return applyWithSchema(record);
+        }
+    }
+
+    private R applySchemaless(R record) {
+        final Map<String, Object> value = (Map<String, Object>) record.value();
```

Comment:
Should we also check the type (String, byte[])  before casting I think we have Map only in the case of JSON deserializer